lib/internal/repl/utils.js

```diff
@@ -392,8 +400,223 @@ function setupPreview(repl, contextSymbol, bufferSymbol, active) {
   return { showPreview, clearPreview };
 }
 
+function setupReverseSearch(repl) {
+  // Simple terminals can't use reverse search.
+  if (process.env.TERM === 'dumb') {
+    return { reverseSearch() { return false; } };
+  }
+
+  const alreadyMatched = new Set();
+  const labels = {
+    r: 'bck-i-search: ',
```

Comment:
Can these prompts be less abbreviated?

Reply:
Is it OK to land this as is? We can also change the name later on in a patch.